cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/TableSegmentRecoveryCommand.java

```diff
@@ -0,0 +1,282 @@
+            }
+        }
+    }
+
+    /**
+     * Given an input {@link ByteArraySegment}, this method scans it for valid Table Segment operations (e.g., PUT, DELETE).
+     * It returns the amount of unprocessed bytes from the input byteArraySegment (e.g., TableEntry split within a chunk
+     * rollover), so it can be used when processing the next chunk.
+     *
+     * @param byteArraySegment Contents of a main Segment data chunk with Table Entries.
+     * @param tableSegmentOperations List to be filled with {@link TableSegmentOperation} objects according to the Table
+     *                               Entries found in the original data.
+     * @return Number of unprocessed bytes, normally due to a partial entry split across two different chunks.
+     */
+    private int scanAllEntriesInTableSegmentChunks(ByteArraySegment byteArraySegment, List<TableSegmentOperation> tableSegmentOperations) {
+        EntrySerializer serializer = new EntrySerializer();
+        int processedBytes = 0;
+        int unprocessedBytesFromLastChunk = 0;
+        while (processedBytes < byteArraySegment.getLength()) {
+            try {
+                ByteArraySegment slice = byteArraySegment.slice(processedBytes, byteArraySegment.getLength() - processedBytes);
+                EntrySerializer.Header header = serializer.readHeader(slice.getBufferViewReader());
+                // If the header has been parsed correctly, then we can proceed.
+                int valueLength = header.getValueLength() < 0 ? 0 : header.getValueLength(); // In case of a removal, use 0 instead of -1.
+                int totalEntryLength = EntrySerializer.HEADER_LENGTH + header.getKeyLength() + valueLength;
+                byte[] keyBytes = slice.slice(EntrySerializer.HEADER_LENGTH, header.getKeyLength())
+                        .getReader().readNBytes(header.getKeyLength());
+                byte[] valueBytes = valueLength == 0 ? new byte[0] : slice.slice(EntrySerializer.HEADER_LENGTH + header.getKeyLength(), header.getValueLength())
+                        .getReader().readNBytes(header.getValueLength());
+                // Add the operation to the list of operations to replay later on (PUT or DELETE).
+                tableSegmentOperations.add(valueBytes.length == 0 ?
+                        new DeleteOperation(TableSegmentEntry.versioned(keyBytes, valueBytes, header.getEntryVersion())) :
+                        new PutOperation(TableSegmentEntry.versioned(keyBytes, valueBytes, header.getEntryVersion())));
+                // Full entry read, so reset unprocessed bytes.
+                Preconditions.checkState(unprocessedBytesFromLastChunk < totalEntryLength, "Some bytes are missing to process.");
+                unprocessedBytesFromLastChunk = 0;
+                processedBytes += totalEntryLength;
+                // Show the progress of scanning Table Segment operations.
+                if (tableSegmentOperations.size() % 100 == 0) {
```

Comment:
should we print the name of the chunk here. else it just looks like the percentage has fallen when we move to the next chunk file.

Reply:
I have added the print file name before the execution of this method, so the progress related to that file.